File: php/src/Environment.php
<?php

declare(strict_types=1);

namespace Idkollen\Client;

enum Environment
{
    case Production;
    case Staging;

    public function baseUrl(): string
    {
        return match ($this) {
            self::Production => 'https://api.idkollen.se',
            self::Staging => 'https://stgapi.idkollen.se',
        };
    }
}

File: csharp/src/Idkollen.Client/Environment.cs
namespace Idkollen.Client;

public enum Environment
{
    Production,
    Staging,
}

internal static class EnvironmentExtensions
{
    public static string BaseUrl(this Environment env) => env switch
    {
        Environment.Production => "https://api.idkollen.se",
        Environment.Staging => "https://stgapi.idkollen.se",
        _ => throw new ArgumentOutOfRangeException(nameof(env)),
    };
}
